### PR TITLE
spice: darwin support

### PR DESCRIPTION
**Pull request by Atemu**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Atemu** [comment]
[QXL](https://gitlab.com/qemu-project/qemu/-/issues/1284) and [spice-app](https://gitlab.com/qemu-project/qemu/-/issues/1285) seem to be broken but you can manually start qemu with `-spice port=5900,disable-ticketing` and attach `remote-viewer spice://127.0.0.1:5900` to verify that SPICE works.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 198257` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages marked as broken and skipped:</summary>
  <ul>
    <li>alpine-make-vm-image</li>
    <li>cloud-init</li>
    <li>cloud-utils</li>
    <li>cot</li>
    <li>python310Packages.cot</li>
    <li>python310Packages.guestfs</li>
    <li>python39Packages.cot</li>
    <li>python39Packages.guestfs</li>
    <li>qemu_full</li>
    <li>qemu_xen</li>
    <li>qemu_xen-light</li>
    <li>qemu_xen_4_10</li>
    <li>qemu_xen_4_10-light</li>
    <li>zpool-auto-expand-partitions</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>out-of-tree</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>appvm</li>
    <li>colima</li>
    <li>libgovirt</li>
    <li>lima</li>
    <li>qemu</li>
    <li>qemu-utils</li>
    <li>qemu_kvm</li>
    <li>spice</li>
    <li>virt-viewer</li>
  </ul>
</details>

File: pkgs/applications/virtualization/libgovirt/no-version-script-ld-flag.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
+ 	-export-dynamic						\
+-	-no-undefined						\
+-	-Wl,--version-script=$(srcdir)/govirt.sym
++	-no-undefined
+ 
+ libgovirt_la_CFLAGS = \
+ 	$(WARNINGFLAGS_C)
+
+Diff finished.  Fri Oct 28 10:36:38 2022
```
[the remark this answers is not in the document]
Comment (etrunko) [reply pkgs/applications/virtualization/libgovirt/no-version-script-ld-flag.patch:14]: Try latest libgovirt which moved to meson and dropped Makefile support.

[the remark this answers is not in the document]
Comment (Atemu) [reply pkgs/applications/virtualization/libgovirt/no-version-script-ld-flag.patch:14]: We'll switch to meson on the next version bump and drop this patch.

**Pull request merged**